components/movementsensor/wheeledodometry/wheeledodometry.go

```diff
@@ -461,6 +472,7 @@ func (o *odometry) DoCommand(ctx context.Context,
 	if ok {
 		o.shiftPos = reset
 		o.originCoord = geo.NewPoint(0, 0)
+		o.coord = geo.NewPoint(0, 0)
```

Comment:
@JohnN193 @martha-johnston is is better to have the DoCommands update the coordinate as well? or did we want the calculation loop to be totally in control here, same with line 486 in this file?

Reply:
See comment above: https://github.com/viamrobotics/rdk/pull/4121/files#r1651321610

This change fixes a race condition.